pkg/daemon/daemon.go

```diff
@@ -279,7 +279,10 @@ func (dn *Daemon) runLoginMonitor(stopCh <-chan struct{}, exitCh chan<- error) {
 		select {
 		case <-stopCh:
 			return
-		case msg := <-sessionNewCh:
+		case msg, ok := <-sessionNewCh:
+			if !ok {
+				continue
```

Comment:
Maybe's let's just `glog.Infof` and `return` here? If the channel is closed, then there's no need to keep this goroutine running, right? (Though of course I agree in practice, closed means probably rebooting).

A stronger approach might be to close the stop channel or something before rebooting so that goroutines can exit cleanly rather than having to deal with "is this an error or are we just rebooting?".

Reply:
Mmm not sure, closing a channel isn't really asking error or not, I've chosen continue as if this is still not rebooting the stopCh can still kick off and tear down everything if we really want to

Reply:
Yeah, though if all we're doing is waiting for the stop channel, then we might as well not be waiting at all and just exit the goroutine, right?

Reply:
Sure I was just chatting about it, I'm going to change it to return, do you really want a log as well? Reboot already logs about it but I guess i can add one at v4

Reply:
The logging part is mostly so that if we're somehow not rebooting, we'll know why new SSH accesses are not resulting in annotations being added.

What I meant re. errors is, if the channel is closed but it's not because we're rebooting, then something else went wrong and we could e.g. try to resubscribe N times. Hence the idea of killing goroutines cleanly before doing the reboot so we don't have to deal with racy errors we don't care about. Not suggesting doing this now though. :)

Reply:
added glog and return, please check it out!